### PR TITLE
ENH: elaborated datasets page, made it difficult to avoid missing the pointer to datasets.d.o

### PR DESCRIPTION
**Pull request by yarikoptic**

### CODE REVIEW

File: content/pages/datasets.rst [hunk 1]
```diff
@@ -5,16 +5,40 @@ Datasets
 
 DataLad can track any data files that are published on the web. It is able
 to monitor and incorporate modifications made to such data sources over time,
-and to publish entire distributions of datasets created by this automated
-process. The DataLad team maintains data trackers for a number of public data
-portals. The generated datasets are available at http://datasets.datalad.org\.
+and to `publish </features.html#data-sharing>`_ entire distributions of datasets created by this
+`automated process </features.html#for-data-consumers>`_.
```
Comment (aqw) [inline content/pages/datasets.rst]: This link destination has nothing to do with the text linked.

File: content/pages/datasets.rst [hunk 2]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+The DataLad team maintains `data trackers
+<https://github.com/datalad/datalad/tree/master/datalad/crawler/pipelines>`__
+for a number of public data portals.
```
Comment (aqw) [inline content/pages/datasets.rst]: This statement is the beginning of a paragraph, not the end.

File: content/pages/datasets.rst [hunk 3]
```diff
@@ -5,16 +5,40 @@ Datasets
...
 
-At present, this website offers uniform access to over 10TB of scientific
-data. This includes datasets from
+Such automatically generated and some manually created and curated
+datasets were then collated into a
```
Comment (aqw) [inline content/pages/datasets.rst]: This is ongoing; thus the past tense is inaccurate.

File: content/pages/datasets.rst [hunk 4]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+`DataLad superdataset <http://docs.datalad.org/en/latest/glossary.html#superdataset>`_.
+Generated superdataset was then published in its entirety
```
Comment (aqw) [inline content/pages/datasets.rst]: Please use articles, such as "the".

File: content/pages/datasets.rst [hunk 5]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+publicly at http://datasets.datalad.org, thus establishing the "official" DataLad
```
Comment (aqw) [inline content/pages/datasets.rst]: Why is "official" in quotes? If you don't explain it, the user will only be left to assume that the official is unofficial and wonder why it's being mislabeled as official.

File: content/pages/datasets.rst [hunk 6]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+data distribution (datalad resource identifier `/// <http://datasets.datalad.org>`__).  Note that to download data
```
Comment (aqw) [inline content/pages/datasets.rst]: I have been told '///' is shorthand for something. Why are you inflicting a perl-esque '///' on an unconvinced audience? Be kinder, give them the full, intuitive command.

File: content/pages/datasets.rst [hunk 7]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+for some of those datasets (e.g. `///crcns <http://datasets.datalad.org/?dir=/crcns>`_),
+you might still need to provide authentication credentials, but otherwise access to them is
```
Comment (aqw) [inline content/pages/datasets.rst]: The use of "you" is suboptimal and is best avoided when not in tutorial mode.

File: content/pages/datasets.rst [hunk 8]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+completely uniform regardless of the data origin.  DataLad has also aggregated
```
Comment (aqw) [inline content/pages/datasets.rst]: Please use the active voice. "DataLad also aggregates..."

File: content/pages/datasets.rst [hunk 9]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+datasets' `metadata <http://docs.datalad.org/en/latest/metadata.html>`_
```
Comment (aqw) [inline content/pages/datasets.rst]: Comma needed at end of line.

File: content/pages/datasets.rst [hunk 10]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+allowing for efficient `search </features.html#data-discovery>`_.
+
+At present, our official distribution offers uniform access to over
+10TB of scientific data. Here is a sample of dataset sources with their DataLad
```
Comment (aqw) [inline content/pages/datasets.rst]: "Here" is awkward and cheap. Reword sentence to actually say what you mean.

File: content/pages/datasets.rst [hunk 11]
```diff
@@ -5,16 +5,40 @@ Datasets
...
+resource identifiers you could pass to `datalad install
```
Comment (aqw) [inline content/pages/datasets.rst]: Please avoid "you".

"pass" is a vague term for real people.

File: content/pages/datasets.rst (hunk 1 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: how come? that section presents crawler -- that is how we generated all those datasets

File: content/pages/datasets.rst (hunk 2 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: ok, making it a leading one for the following paragraph

File: content/pages/datasets.rst (hunk 3 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: ok, thanks

File: content/pages/datasets.rst (hunk 4 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: thansk

File: content/pages/datasets.rst (hunk 5 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: ;-) well, I guess I add too much value into a typical use of `official` word... removed quotes

File: content/pages/datasets.rst (hunk 6 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: because I reference/use it later in the text.

My idea, may be not formalized the best way, is that as audience goes through the pages of the website they could get increasing number of terms/functions and pointers  into details of the datalad.  
Majority of audience haven't heard about `perl` probably yet so wouldn't yet be that biased ;)

File: content/pages/datasets.rst (hunk 7 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: ok rephrased to "it might be necessary to"

File: content/pages/datasets.rst (hunk 8 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: ok

File: content/pages/datasets.rst (hunk 9 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: done

File: content/pages/datasets.rst (hunk 10 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: rephrased to use your original form

File: content/pages/datasets.rst (hunk 11 above)
Comment (yarikoptic) [reply content/pages/datasets.rst]: replaced with "for use with"

**Comment by aqw** [comment]
@yarikoptic Normally I am happy to do a quick merge and just edit on top. Michael and I often work that way and you and I can do that as well if you're comfortable with it. I didn't want you to view that as hostile or anything.

**Comment by yarikoptic** [comment]
merge + edit is ok with me as well, but then it is a bit harder to review the changes since would need to fetch go into gitk or smth etc.  pushing directly into PR we could then together converge it to the point of agreement and be done

also this flow is better in cases when there is that automatic deployment I have mentioned in #25 -- as publishing would happen from master branch, it is better to converge to a solution in a PR and then cause automated publishing of the content upon merge.  Also, it would guarantee that master "does not fail" (to build/test/etc) as it could be the case with changes with merge + fixup flow.

**Comment by aqw** [comment]
@yarikoptic Ok, just to be clear (because I'm new to this proposed workflow):

* you open PR
* I note changes that are needed
* I checkout your PR's branch
* I commit my changes and push to (or PR to?) your branch
* so that way your PR now has your commits and mine

Yes?

Or am I missing something.

---Alex

**Comment by yarikoptic** [comment]
Exactly!  
Generally you can push directly into my branch, unless you feel that changes require my review (your judgment). If later --  then you could send a PR against my clone, and if no further changes needed I would merge the PR and it would also appear within the original PR.

FWIW, I have a little helper to checkout PR branches (http://git.onerussian.com/?p=etc/bash.git;a=blob;f=.bash/bashrc/30_aliases_sh;h=28fa4ad215839110953f598becbbbd586ff8edb0;hb=HEAD#l902), but for constant collaborators I prefer just adding their repos as remotes (another helper I use is `ghremote` I have included in https://github.com/datalad/datalad/blob/master/CONTRIBUTING.md), fetching them and checking out those branches.

**Comment by aqw** [comment]
@yarikoptic Thanks for the clarification. I'll give that a try.

---Alex

**Comment by yarikoptic** [comment]
Thanks in advance @aqw

File: content/pages/datasets.rst (hunk 1 above)
Comment (aqw) [reply content/pages/datasets.rst]: My bad; I must have had a brain fart. I didn't realize the crawler info had been put under the "Data Consumers" feature heading. I thought that heading contains the info that I now see is under "Data Discovery".

You're right; this link name is accurate.

File: content/pages/datasets.rst (hunk 6 above)
Comment (aqw) [reply content/pages/datasets.rst]: Then I need to ask the obvious followup... couldn't we just change it elsewhere in the text too? ;-) Or is the full, intuitive form of `///` just that awkward and unwieldy?

I guess I'm trying to understand why the text avoids using the long form (that I assume is clear) and instead attempts to convince the user that "this is easy" by describing a "datalad resource identifier" using a stack of punctuation that has no apparent meaning.

**Comment by aqw** [comment]
@yarikoptic I made some changes. And I hope I kept the spirit and direction of where you were going. Let me know if/how you feel I strayed from that. I don't have as much experience writing from your headspace. It'll take time. :-)

I'm unhappy with how I shoe-horned the "be automated" link in there, but I had grown attached to the wording of the paragraph and havn't come up yet with a cleaner pair of words to link.

---Alex

**Comment by yarikoptic** [comment]
Looks great. Thanks
Since we are both satisfied - merging :-)

**Comment by aqw** [comment]
@yarikoptic Sounds good. Found a small typo I made. Will fix.

Quick question: do you/the-datalad-project prefer merging PRs so that the name of the branch is retained in the git history? Or are you also ok with a `curl github-url.patch | git am` workflow (which is what I usually use when merging PRs). Just making sure I'm learning all the angles of this new workflow.

---Alex

**Comment by yarikoptic** [comment]
I think we never agreed on anything particular about merge messages, so `git am` workflow should be perfectly fine I think.  thanks for asking

**Comment by aqw** [comment]
@yarikoptic Thanks for the clarification. I usually take the `git am` approach because I have this irrational aversion to the merge lines in tig's history. ;-)

---Alex

**Comment by yarikoptic** [comment]
note though, that if you were talking about "merging" entire PRs with multiple commits, `curl URLtodiff | git am` (not really a merge but just a commit) approach would probably be not the best way since might loose information about original/multiple authors and informative commit messages along the development line.  And in general I like seeing those branches being developed in their parallel lines to the master and then merged.
But you are the lead of the website work here so it is for you to find a good compromise which would fit first of all your workflow ;)  I could adjust

**Comment by aqw** [comment]
@yarikoptic https://github.com/datalad/datalad.org/pull/24.patch doesn't retain multi-author info and non-subject commit messages? I've never noticed a problem, but I'll definitely double-check that, because if that's true, I'll certainly abandon the `git am` method.

---Alex

**Comment by yarikoptic** [comment]
Oh no, I was wrong apparently - I just assumed that all changes would be squashed into one commit. Still Not sure what is the advantage really, but if you like it this way - works for me

**Comment by aqw** [comment]
@yarikoptic Whew, ok, good.

Other than my joking aversion to branch-lines in tig, using `git am` has one advantage to me. I can interactively rebase commits prior to merging them. In a different project I maintain, I frequently receive patches from inexperienced contributors. If there's only small nitpick-fixes to be made, I offer the contributor the opportunity to fix them, otherwise I tell them I'm happy to do it for them. Often they choose the latter. So I `git am` it in, make my changes, squash them in as a fix, usually cleanup the commit subject so it's actually useful, and then add "close #issue-num" to the commit message. Then it's ready to be pushed.

Basically, `git am` makes it easy the rebase a PR into a more ready condition. It's a few more steps to do that with a merge (add remote, fetch, checkout, make changes, squash, then merge/cherry-pick).

That's the rational. I'm not super attached to it; it's just been my workflow for a few years now.

---Alex

**Comment by yarikoptic** [comment]
I cannot blame you in wrong-doing whenever you seems to be using the workflow of the linux/git gods ;)

**Comment by aqw** [comment]
Hah. Good to know. I had no idea that was Linux's preferred workflow. :-)

---Alex
